From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sleigh-InSPECtor <sleighinspector@outlook.com>
Date: Thu, 16 May 2024 15:06:26 +0930
Subject: [PATCH] 6531: AArch32: vsel "gt" condition missing NG & OV flag
 equality check

AArch32: fixed vselgt

* added 'and N == V' check missed in the vselcond "gt"
---
 Ghidra/Processors/ARM/data/languages/ARMv8.sinc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Processors/ARM/data/languages/ARMv8.sinc b/Ghidra/Processors/ARM/data/languages/ARMv8.sinc
index 79ff865b16..ab4d26b37d 100644
--- a/Ghidra/Processors/ARM/data/languages/ARMv8.sinc
+++ b/Ghidra/Processors/ARM/data/languages/ARMv8.sinc
@@ -1327,7 +1327,7 @@ vselcond: "ge"
 vselcond: "gt"
 	is ((TMode=0 &     c2021=0b11)
 	|   (TMode=1 & thv_c2021=0b11))
-	{ tmp:1 = (!ZR); export tmp; }
+	{ tmp:1 = (!ZR && NG==OV); export tmp; }
 vselcond: "vs"
 	is ((TMode=0 &     c2021=0b01)
 	|   (TMode=1 & thv_c2021=0b01))
-- 
2.45.0

